### PR TITLE
Deprecated no-reserved-keywords

### LINKED ISSUE #652: Mark no-reserved-keywords as deprecated
### Bug Report
 
-   **`tslint-microsoft-contrib` version**: 6.0.0-beta
-   **TSLint version**: 5.11
-   **TypeScript version**: 3.1.3
-   **Running TSLint via**: *

There are still issues being reported with `no-reserved-keywords` (#395, #403), but the rule is an inferior version of TSLint core's [`variable-name`](https://palantir.github.io/tslint/rules/variable-name) with `ban-keywords`, right?

If we could remove `no-reserved-keywords`, that would be great.

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
#### PR checklist

-   [x] Addresses an existing issue: fixes #652
-   [x] New feature, bugfix, or enhancement
    -   [x] Includes tests
-   [x] Documentation update

#### Overview of change:

Marks the rule as deprecated per #652.

### CODE REVIEW

File: tslint-warnings.csv [hunk 1]
```diff
@@ -262,9 +262,9 @@ react-a11y-no-onchange,"For accessibility of your website, enforce usage of onBl
...
-react-a11y-role-supports-aria-props,Enforce that elements with explicit or implicit roles defined contain only `aria-*` properties supported by that `role`.,TSLINT1IFADTQ,tslint,Non-SDL,Warning,Important,Opportunity for Excellence,See description on the tslint or tslint-microsoft-contrib website,TSLint Procedure,,
+react-a11y-role-supports-aria-props,"Enforce that elements with explicit or implicit roles defined contain only `aria-*` properties supported by that `role`.Many aria attributes (states and properties) can only be used on elements with particular roles.Some elements have implicit roles, such as `<a href='hrefValue' />`, which will be resolved to `role='link'`.A reference for the implicit roles can be found at [Default Implicit ARIA Semantics](https://www.w3.org/TR/html-aria/#sec-strong-native-semantics).",TSLINT1IFADTQ,tslint,Non-SDL,Warning,Important,Opportunity for Excellence,See description on the tslint or tslint-microsoft-contrib website,TSLint Procedure,,
```
Comment (JoshuaKGoldberg) [inline tslint-warnings.csv]: Unrelated. If someone else sends a PR that happens to include them, they should go away here. Seems fine to keep in.

Comment (IllusionMH) [reply tslint-warnings.csv]: Why there are no spaces after `.`? Like ``that `role`.Many``, `particular roles.Some `, ``to `role='link'`.A reference``.

Change above contain spaces (to many?). Probably requires build script update.

Comment (JoshuaKGoldberg) [reply tslint-warnings.csv]: 😄 I'll go ahead and file a separate issue & extract from this PR...

Comment (JoshuaKGoldberg) [reply tslint-warnings.csv]: #675

**Comment by astorije** [comment]
:shipit: :shipit: :shipit: 

@JoshuaKGoldberg, are you planning on a release today? (Just to know if I temporarily `tslint:disable` in my code or not :D)

**Comment by JoshuaKGoldberg** [comment]
> :shipit: :shipit: :shipit:

Lol, forgot about that piece of history...

I wasn't planning on it, since this is the only user-facing change that goes in it. It's probably better to just disable `no-reserved-keywords`. Sorry!

**Comment by astorije** [comment]
Not even worth a teeeeny tiny v6.0.1? 🙏😁
No worries if not.

**Comment by JoshuaKGoldberg** [comment]
Haha sorry no, it'd be a breaking change.

**Comment by astorije** [comment]
Really? I didn't realize it was a breaking change, I thought that was a simple deprecation for now.

**Comment by JoshuaKGoldberg** [comment]
The added `console.warn` can be surprisingly confusing. I had a coworker new to the area completely disable TSLint locally because we'd bumped to a version and forgotten to disable `no-unused-variables` and they'd interpreted the warning as TSLint erroring out. It took us a little while to realize why they had so many "fix lint" commits in their PRs!

**Comment by astorije** [comment]
That makes sense, thanks!

**Pull request merged**